src/snappy.c

```diff
@@ -126,7 +126,55 @@ static inline bool is_little_endian(void)
 	return false;
 }
 
-#ifndef _MSC_VER
+#if defined(__xlc__)
+#define rd_clz(n)   __cntlz4(n)
+#define rd_ctz(n)   __cnttz4(n)
+#define rd_ctz64(n) __cnttz8(n)
+
+#elif defined(__SUNPRO_C)
+// Source for following defintions is Hacker’s Delight, Second Edition by Henry S. Warren
```

Comment:
Fixed typo. I have added the link to use content which says : "You are free to use, copy, and distribute any of the code on this web site, whether modified by you or not. You need not give attribution. "